ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/CommitContext.scala

```diff
@@ -8,16 +8,19 @@ import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
   DamlStateKey,
   DamlStateValue
 }
-import com.daml.ledger.participant.state.kvutils.{Err, DamlStateMap}
+import com.daml.ledger.participant.state.kvutils.{DamlStateMap, Err}
 import com.daml.ledger.participant.state.v1.ParticipantId
 import com.daml.lf.data.Time.Timestamp
+import org.slf4j.LoggerFactory
 
 import scala.collection.mutable
 
 /** Commit context provides access to state inputs, commit parameters (e.g. record time) and
   * allows committer to set state outputs.
   */
 private[kvutils] trait CommitContext {
+  private[this] val logger = LoggerFactory.getLogger(this.getClass)
+
   def inputs: DamlStateMap
   // NOTE(JM): The outputs must be iterable in deterministic order, hence we
   // keep track of insertion order.
```

Comment:
Every `Committer` uses its own `CommitContext`, so the output order is determinized for each DamlSubmission in isolation. The output order across several DamlSubmissions does not seem to be deterministic given that `BatchedSubmissionValidator` calls `commit`s inside a `mapAsync`. I'm not sure though why we even need deterministic output order here. Can you check why we need the deterministic output order in the first place?

Reply:
We ensure determinisim in `BatchedSubmissionValidator` by doing the conflict detection first (according to a fixed order) and only `commit`ing submissions that won't conflict and will generate the same write set regardless of order of execution.
We need deterministic output for historical reasons here -- we may want to relax this requirement.

Reply:
Makes sense. Thanks for explaining the historical reasons.